src/py/flwr/cli/flower_toml.py

```diff
@@ -0,0 +1,153 @@
+# Copyright 2024 Flower Labs GmbH. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Utility to validate the flower.toml file.
+
+This module will be improved over time.
+"""
+
+import importlib
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import tomli
+
+
+def load_flower_toml(path: Optional[str] = None) -> Optional[Dict[str, Any]]:
+    """Load flower.toml and return as dict."""
+    if path is None:
+        cur_dir = os.getcwd()
+        toml_path = os.path.join(cur_dir, "flower.toml")
+    else:
+        toml_path = path
+
+    if not os.path.isfile(toml_path):
+        return None
+
+    with open(toml_path, encoding="utf-8") as toml_file:
+        data = tomli.loads(toml_file.read())
+        return data
+
+
+def validate_flower_toml_fields(config: Dict[str, Any]) -> Tuple[bool, List[str]]:
+    """Validate flower.toml fields."""
+    invalid_reasons = []
+
+    if "project" not in config:
+        invalid_reasons.append("Missing [project] section")
+    else:
+        if "name" not in config["project"]:
+            invalid_reasons.append('Property "name" missing in [project]')
+        if "version" not in config["project"]:
+            invalid_reasons.append('Property "version" missing in [project]')
+        if "description" not in config["project"]:
+            invalid_reasons.append('Property "description" missing in [project]')
+        if "license" not in config["project"]:
+            invalid_reasons.append('Property "license" missing in [project]')
```

Comment:
Do we want to force people to add a license section to their private projects? It's not mandatory in `pyproject.toml`

Reply:
@danieljanes theoretically only `name` and `version` [is required](https://packaging.python.org/en/latest/specifications/pyproject-toml/#declaring-project-metadata-the-project-table). We can keep close to the spec adhere to that. Shall I remove the requirements?

Reply:
@danieljanes I can add a test that checks that only allowed keys are present as described in the spec. I'd not implement the key level requirements right now. We can improve on that later on.

Reply:
@tanertopal how about this: we required only `name` and `version` to be in line with the spec. In addition to that, we log warnings for the other fields that we consider to be important.

Reply:
Sounds good.

Reply:
@danieljanes warnings are implemented